PsfShimMonitor/MainWindow.xaml.cs

```diff
@@ -0,0 +1,720 @@
+
+                    try
+                    {
+                        operation = (string)data.PayloadByName("Operation");
+                    }
+                    catch { }
+                    try
+                    {
+                        inputs = (string)data.PayloadByName("Inputs");
+                    }
+                    catch { }
+                    try
+                    {
+                        result = (string)data.PayloadByName("Result");
+                    }
+                    catch { }
+                    try
+                    {
+                        outputs = (string)data.PayloadByName("Outputs");
+                    }
+                    catch { }
+                    try
+                    {
+                        caller = (string)data.PayloadByName("Caller");
+                    }
+                    catch { }
+                    if (inputs == null & result == null && outputs == null)
+                    {
+                        try
+                        {
+                            outputs = (string)data.PayloadByName("Message");
+                        }
+                        catch { }
+                    }
+                    try
+                    {
+                        start = (Int64)data.PayloadByName("Start");
+                    }
+                    catch { }
+                    try
+                    {
+                        end = (Int64)data.PayloadByName("End");
+                    }
+                    catch { }
+                    EventItem ei = new EventItem((int)data.EventIndex,
+                                                    start,
+                                                    end,
+                                                    data.TimeStamp,
+                                                    data.ProcessName,  
+                                                    data.ProcessID,
+                                                    data.ThreadID,
+                                                    data.ProviderName,
+                                                    operation,
+                                                    inputs,
+                                                    result,
+                                                    outputs,
+                                                    caller
+                                                  );
+
+                    lock (_TEventListsLock)
+                    {
+                        _TEventListItems.Add(ei);
+                        AddToProcIDsList(data.ProcessID);
+                    }
+                    worker.ReportProgress((int)data.EventIndex);
+                };
+
+                res1 = myTraceEventSession.EnableProvider(etwp.guid);    
+                if (!res1)
+                {
+                    // Attempt resetting for second run...
+                    myTraceEventSession.DisableProvider(etwp.guid);
```

Comment:
You are restarting for a second run.  Is 1 restart adequate?

Reply:
It seems to be.  While the initial call can fail if a previous attempt didn't shut down nicely, just trying to enable seems to clear up the issue and the second attempt succeeds.  There doesn't seem to be any clear documentation on this.  If the second attempt were to fail, it doesn't make sense to continue on (a reboot might be in order).  The code logic would display the error in the status area.